client/layout/index.jsx

```diff
@@ -222,10 +222,20 @@ class Layout extends Component {
 		this.refreshColorScheme( undefined, this.props.colorScheme );
 	}
 
+	/**
+	 * Refresh the color scheme if
+	 * - the color scheme has changed
+	 * - the global sidebar is visible and the color scheme is not `modern`
+	 * - the global sidebar was visible and is now hidden and the color scheme is not `modern`
+	 * @param prevProps object
+	 */
 	componentDidUpdate( prevProps ) {
 		if (
 			prevProps.colorScheme !== this.props.colorScheme ||
-			( this.props.isGlobalSidebarVisible && this.props.colorScheme !== 'modern' )
+			( this.props.isGlobalSidebarVisible && this.props.colorScheme !== 'modern' ) ||
```

Comment:
Not necessary for this PR, but a possible improvement we could / probably should make here. Define a `const GLOBAL_COLOR_SCHEME = 'modern';` and use that everywhere replacing 'modern' with that constant where we are using it for the context of the global color scheme.